### PR TITLE
Avoid initializing SwiftASTContext when looking up Clang names. (NFC)

### PR DESCRIPTION
**Pull request by adrian-prantl**
With this change we can set a breakpoint on
SwiftASTContext::SwiftASTContext() in a non-asserts LLDB and run
TestSwiftFoundationTypeURL and the breakpoint will not trigger.  We do
this by initializing an empty swift::ASTContext, whose sole purpose is
to host a a ClangImporter's imported DeclNames. Initializing it is
safe & fast, since no modules are being imported.

This is a much simpler but just as effective alternative to https://github.com/apple/swift/pull/41561 and https://github.com/apple/llvm-project/pull/4011

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h [hunk 1]
```diff
@@ -25,6 +25,8 @@
 
 namespace swift {
 class DWARFImporterDelegate;
+class DiagnosticEngine;
+class SourceManager;
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h]: why are these needed in the header?

**Review by kastiglione — APPROVED** [review]
much simpler indeed

**Comment by adrian-prantl** [comment]
@swift-ci test
